flink-libraries/flink-table/src/main/scala/org/apache/flink/api/java/table/TableEnvironment.scala

```diff
@@ -75,5 +75,15 @@ class TableEnvironment {
       TypeExtractor.createTypeInfo(clazz).asInstanceOf[TypeInformation[T]])
   }
 
+  /**
+   * Converts the given [[org.apache.flink.api.table.Table]] to
+   * a DataSet. The given type must have exactly the same fields as the
+   * [[org.apache.flink.api.table.Table]]. That is, the names of the
```

Comment:
I would make name equivalence only required for POJOs and generic composite types types. 
Rows and tuples can be matched by position. Otherwise, fields would need to be renamed to `f0`, `f1`, etc. for tuples.